src/connector/src/sink/encoder/template.rs

```diff
@@ -95,3 +150,181 @@ impl RowEncoder for TemplateEncoder {
         Ok(s)
     }
 }
+
+pub struct TemplateRedisGeoValueEncoder {
+    schema: Schema,
+    col_indices: Option<Vec<usize>>,
+    lat_col: usize,
+    lon_col: usize,
+}
+
+impl TemplateRedisGeoValueEncoder {
+    pub fn new(
+        schema: Schema,
+        col_indices: Option<Vec<usize>>,
+        lat_name: &str,
+        lon_name: &str,
+    ) -> Result<Self> {
+        let lat_col = schema
+            .names_str()
+            .iter()
+            .position(|name| name == &lat_name)
+            .ok_or_else(|| {
+                SinkError::Redis(format!("Can't find lat column({}) in schema", lat_name))
+            })?;
+        let lon_col = schema
+            .names_str()
+            .iter()
+            .position(|name| name == &lon_name)
+            .ok_or_else(|| {
+                SinkError::Redis(format!("Can't find lon column({}) in schema", lon_name))
+            })?;
+        Ok(Self {
+            schema,
+            col_indices,
+            lat_col,
+            lon_col,
+        })
+    }
+
+    pub fn encode_cols(
+        &self,
+        row: impl Row,
+        _col_indices: impl Iterator<Item = usize>,
```

Comment:
Perhaps off-the-topic but the trait function is confusing to me: why `col_indices` need to be passed from outside, while it's stored in `self`? 😂

Reply:
cc @xiangjinwu